### PR TITLE
WIP: Add the cinder serverspec

### PR DESCRIPTION
**Pull request by vlope093**
Dependent on PR #1554

### CODE REVIEW

File: roles/cinder-common/templates/etc/serverspec/cinder_spec.rb [hunk 1]
```diff
@@ -0,0 +1,45 @@
+require 'spec_helper'
+
+files = {"cinder.conf"=> 640, "api-paste.ini"=> 640, "policy.json"=> 644, "rootwrap.conf" =>640}
```
Comment (j2sol) [inline roles/cinder-common/templates/etc/serverspec/cinder_spec.rb]: inconsistent whitespace on this line.

File: roles/cinder-common/templates/etc/serverspec/cinder_spec.rb [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+files.each do | file, mode|
+  
+#describes what are we going to test
+  describe file("/etc/cinder/#{file}") do
+    it { should be_mode mode}
+    it { should be_owned_by 'cinder' } 
+    it { should be_grouped_into'cinder' }
```
Comment (j2sol) [inline roles/cinder-common/templates/etc/serverspec/cinder_spec.rb]: whitespace

File: roles/cinder-common/templates/etc/serverspec/cinder_spec.rb [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+#describes what are we going to test
+  describe file("/etc/cinder/#{file}") do
+    it { should be_mode mode}
+    it { should be_owned_by 'cinder' } 
+    it { should be_grouped_into'cinder' } 
+    
+  end
+end 
+
+files = {"cinder-api.log" => 'cinder', "cinder-volume.log" => 'cinder', "cinder-scheduler.log" => 'cinder' ,"cinder-manage.log" => 'root'}
```
Comment (j2sol) [inline roles/cinder-common/templates/etc/serverspec/cinder_spec.rb]: Inconsistent whitespace.

File: roles/cinder-common/templates/etc/serverspec/cinder_spec.rb [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  describe file("/var/log/cinder/#{file}") do
+    it { should be_mode 644 }
+    it { should be_owned_by owner }
+    it { should be_grouped_into 'adm' }
+  end
+end
+
+describe file('/etc/logrotate.d/cinder') do
+  file_contents = [ '# Generated by Ansible.',
+           '# Local modifications will be overwritten.',
```
Comment (j2sol) [inline roles/cinder-common/templates/etc/serverspec/cinder_spec.rb]: Whitespace looks weird on this line.

File: roles/cinder-common/templates/etc/serverspec/cinder_spec.rb [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+         '  compress',
+         '}']
+  file_contents.each do |file_line|
+    it { should contain file_line}
+  end
+end
+
+describe file('/etc/cinder/cinder.conf') do
+  it {should contain 'log_dir = /var/log/cinder'}
+  it {should contain 'debug = True'}
```
Comment (j2sol) [inline roles/cinder-common/templates/etc/serverspec/cinder_spec.rb]: Debug value should derive from `cinder.logging.debug`.

Comment (pwplusnick) [inline roles/cinder-common/templates/etc/serverspec/cinder_spec.rb]: make that line
`  it { should contain 'debug = {{ cinder.logging.debug }}' }`

File: roles/cinder-common/templates/etc/serverspec/cinder_spec.rb [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+files = { "cinder.conf"=> 640, "api-paste.ini"=> 640, "policy.json"=> 644, "rootwrap.conf"=> 640 }  
+files.each do |file, mode|
+  describe file("/etc/cinder/#{file}") do
+    it { should be_mode mode }
+    it { should be_owned_by 'cinder' } 
+    it { should be_grouped_into'cinder' } 
+  end
+end 
+
+files = {"cinder-api.log"=> 'cinder', "cinder-volume.log" => 'cinder', "cinder-scheduler.log" => 'cinder' ,"cinder-manage.log" => 'root'}
```
Comment (j2sol) [inline roles/cinder-common/templates/etc/serverspec/cinder_spec.rb]: Still some inconsistent whitespace on this line.

**Comment by j2sol** [comment]
This looks mostly right now. Just a bit of whitespace clean up and then squash all the commits into one and it'll be good to go once tests pass.

**Comment by j2sol** [comment]
ok to test

**Comment by j2sol** [comment]
THe bits in cinder depend on things done in the serverspec role, and the config defaults from the server spec role. Looks like we need to make the cinder role depend on the serverspec role.

Or lose the conditional? This needs more thought out.

**Comment by j2sol** [comment]
retest

File: roles/cinder-common/tasks/meta/main.yml [hunk 7]
```diff
@@ -0,0 +1,3 @@
+---
```
Comment (j2sol) [inline roles/cinder-common/tasks/meta/main.yml]: This location is wrong. It should be in `roles/cinder-common/meta/main.yml`

**Comment by j2sol** [comment]
I have no idea why this is failing.

**Comment by j2sol** [comment]
retest

File: roles/cinder-common/meta/main.yml [hunk 8]
```diff
@@ -22,3 +22,4 @@ dependencies:
     when: openstack_install_method == 'package'
   - role: ceph-common
     when: ceph.enabled|bool
+    role: serverspec
```
Comment (pwplusnick) [inline roles/cinder-common/meta/main.yml]: This needs to be `- role: serverspec`. The role serverspec is not currently part of the yaml list.

**Comment by j2sol** [comment]
retest

**Comment by j2sol** [comment]
retest

**Comment by pwplusnick** [comment]
@j2sol: @vlope093: It failed to get some of the python dependencies. I would try another retest because I doubt these changes would effect the common role at all.

**Pull request merged**